consensus/common_test.go

```diff
@@ -306,26 +307,97 @@ func randConsensusState(nValidators int) (*ConsensusState, []*validatorStub) {
 
 //-------------------------------------------------------------------------------
 
-func ensureNoNewStep(stepCh <-chan interface{}) {
-	timer := time.NewTimer(ensureTimeout)
+func ensureNoNewEvent(t *testing.T, ch <-chan interface{}, timeout time.Duration,
+	errorMessage string) {
 	select {
-	case <-timer.C:
+	case <-time.After(timeout):
 		break
-	case <-stepCh:
-		panic("We should be stuck waiting, not moving to the next step")
+	case <-ch:
+		t.Error(errorMessage)
```

Comment:
We probably want these to be panics for now until we build something more intelligent, otherwise we won't know which line in the actual test function caused the failure.

Reply:
Or alternatively to the error check in the actual test and `t.Fatalf` there. I know some may be concerned about readability but it's a test after all and the execution should be clear.

Reply:
For now, I would also say to leave panic. We will improve it after.